src/forms/text.rs

```diff
@@ -0,0 +1,387 @@
+pub struct Text {
+    screen: Screen,
+    dirty: bool,
+    active: bool,
+    input_buffer: String,
+    cursor_pos: usize,
+    options: FormOptions,
+}
+
+impl Text {
+    pub fn new(w: u32, options: FormOptions) -> Self {
+        Self {
+            screen: Screen::new(w, 1),
+            dirty: true,
+            active: false,
+            input_buffer: String::new(),
+            cursor_pos: 0,
+            options,
+        }
+    }
+
+    /// Sets a specific value inside the field
+    pub fn set_input_buffer(&mut self, input: &str) {
+        self.dirty = true;
+        self.input_buffer = String::from(input);
+        self.move_cursor(i32::MAX);
+    }
+
+    /// Clear the field
+    pub fn clear_input_buffer(&mut self) {
+        self.dirty = true;
+        self.input_buffer = String::new();
+        self.cursor_pos = 0;
+    }
+
+    /// Insert a character at the position of the cursor
+    pub fn put_char(&mut self, chr: char) {
+        let mut new_buffer = self.input_buffer[0..self.cursor_pos].to_string();
+        new_buffer.push(chr);
+        new_buffer.push_str(&self.input_buffer[self.cursor_pos..]);
+        self.input_buffer = new_buffer;
+        self.move_cursor(1);
+    }
+
+    /// Removes a certain amount of characters either on the left (positive) or right (negative) side of the cursor
+    pub fn remove_char(&mut self, amount: i32) {
+        if amount == 0 {
+            return;
+        }
+        self.dirty = true;
+        let off_l = amount.max(0) as usize; // offset to the left from cursor, `positive` or 0
+        let off_r = amount.min(0).abs() as usize; // offset to the right from cursor,  `-negative` or 0
+        let pos_l = self.cursor_pos.saturating_sub(off_l);
+        let pos_r = self
+            .cursor_pos
+            .saturating_add(off_r)
+            .min(self.input_buffer.len());
+        self.input_buffer =
+            String::from(&self.input_buffer[0..pos_l]) + &self.input_buffer[pos_r..]; // this skips the cursor +/- offsets
+        self.move_cursor(-amount.max(0));
+    }
+
+    /// Moves the cursor left (negative) or right (positive)
+    ///
+    /// The cursor is clamped at its boundaries
+    pub fn move_cursor(&mut self, amount: i32) {
+        self.dirty = true;
+        self.cursor_pos = (self.cursor_pos as i64 + amount as i64)
+            .clamp(0, self.input_buffer.len() as i64) as usize;
+    }
+}
+
+impl FormField for Text {
+    fn make(w: u32, options: FormOptions) -> Self
+    where
+        Self: Sized,
+    {
+        Self::new(w, options)
+    }
+
+    fn reset(&mut self) {
+        self.clear_input_buffer();
+    }
+
+    fn get_width(&self) -> u32 {
+        self.screen.get_width()
+    }
+
+    fn get_height(&self) -> u32 {
+        self.screen.get_height()
+    }
+
+    fn resize(&mut self, w: u32, _h: u32) {
+        self.dirty = true;
+        self.screen.resize(w, 1);
+    }
+
+    fn handle_event(&mut self, event: &Event) {
+        if self.active {
```

Comment:
Personally I'd change this to:
```rust
if !self.active {
    return;
}
...
```
or alternatively only send events to those FormFields that are active.